Create a blog entry for 0.11 and Melange (#743)
reasonml/reason-react: website/blog/2023-06-11-reborn.md

---

> title: ReasonReact is back (Melange from now on)

Reviewer: ```suggestion
title: ReasonReact is back (as a library for Melange from now on)
```
?

Author: `title: ReasonReact is back (with Melange from now on)`

Reviewer: Sounds good. Could also be `ReasonReact is back (now with Melange)`.

---

Since the ReScript project diverged from ReasonML and moved ReasonReact into `@rescript/react`, ReasonReact has been in limbo. While this happened, the Melange team released the first major release of [Melange](https://buttondown.email/anmonteiro/archive/melange-hits-v10/) and made the first steps towards a deeper integration with the OCaml ecosystem.

We are happy to announce that ReasonReact is back and will be maintained and developed by the Melange team.

## What does this mean

We released ReasonReact to the [opam registry](https://opam.ocaml.org/packages/reason-react) and be installed with opam or esy if you prefer (even though Melange team recommends using opam).

We updated the documentation to reflect these new changes and we will continue to improve it.

> ## Changes

Author: It's mean to be released for 0.12, so I prefer to keep it for later

The previous version published in npm compatible with ReScript is 0.9.0. Between 0.9.0 and 0.11.0, we've fixed a few things:

- [0.9.2] Add drag event API binding
- [0.9.2] Add `aria-haspopup` as ariaHaspopup as prop
- [0.9.2] Add `aria-current` as ariaCurrent as prop
- [0.9.2] Add `aria-checked` as ariaChecked as prop
- [0.10.0] Deprecate legacy modules (ReasonReact, ReasonReactCompat, ReasonReactOptimizedCreateClass)

### Version 0.11

- [0.11.0] Add suppressHydrationWarning
- [0.11.0] Created opam package (reason-react.opam)
- [0.11.0] Requires dune 3.8.0 (via dune-project)
- [0.11.0] Full migration to Melange

## Migration

First make sure your project builds with Melange, once this is done, you can follow the migration guide:

- Install latest version of ReasonReact: `opam install reason-react`

> - Use this as a package in the dune file `(libraries reason-react)` (make sure to add `react-jsx-ppx` on the pps as well).

Reviewer: Is it worth adding more explanation? e.g. 
```suggestion
- Use this as a library in the dune file `(libraries reason-react)` (make sure to add `(preprocess (pps reason-react-ppx))` as well, if you use Reason syntax).
```

> - Remove usages of modules `ReactDOMRe`, `ReactEventRe`, `ReasonReact`, `ReasonReactCompat`, `ReasonReactOptimizedCreateClass`. They were deprecated a long time ago, but we removed them in this version. There should be a 1:1 mapping between the old modules and the new ones.

Reviewer: ```suggestion
- Replace usages of modules with their newer names: `ReactDOMRe` -> `ReactDOM`, `ReactEventRe` -> `ReactEvent`, `ReasonReact` -> `React`
- Remove usages of `ReasonReactCompat` and `ReasonReactOptimizedCreateClass`. They were deprecated a long time ago, and we removed them in this version.
```

---

> title: ReasonReact is back (now with Melange)

Reviewer 2: thoughts on this alternative title?
```suggestion
title: ReasonReact is back (and powered by Melange)
```

---

> Since the ReScript project diverged from ReasonML and moved ReasonReact into `@rescript/react`, ReasonReact has been in limbo. While this happened, the Melange team released the first major release of [Melange](https://buttondown.email/anmonteiro/archive/melange-hits-v10/) and made the first steps towards a deeper integration with the OCaml ecosystem.

Reviewer 2: ```suggestion
ReasonReact moved to `@rescript/react` after the ReScript announcement. The future of ReasonReact has been uncertain ever since. In the meantime, the Melange team announced the first major release of [Melange](https://anmonteiro.substack.com/p/melange-10-is-here) and took the first steps towards deeper integration with the OCaml ecosystem.
```

We are happy to announce that ReasonReact is back and will be maintained and developed by the Melange team.

## What this means

We released ReasonReact to the [opam registry](https://opam.ocaml.org/packages/reason-react) and be installed with opam or esy if you prefer (even though Melange team recommends using opam).

We updated the documentation to reflect these new changes and we will continue to improve it over time.

> ## Changes

Reviewer 2: ```suggestion
## Full changelog
```

Reviewer 2: I propose moving this section last.

Once I looked at a "full changelog" I thought it was the last section and I wouldn't read past it.

> The previous version published in npm compatible with ReScript is 0.9.0. Between 0.9.0 and 0.11.0, we've fixed a few things:

Reviewer 2: ```suggestion
The version of ReasonReact published to npm is compatible with ReScript 0.9.0. Since then, we've fixed a few things:
```

- [0.9.2] Add drag event API binding
- [0.9.2] Add `aria-haspopup` as ariaHaspopup as prop
- [0.9.2] Add `aria-current` as ariaCurrent as prop
- [0.9.2] Add `aria-checked` as ariaChecked as prop
- [0.10.0] Deprecate legacy modules (ReasonReact, ReasonReactCompat, ReasonReactOptimizedCreateClass)

### Version 0.11

- [0.11.0] Add suppressHydrationWarning
- [0.11.0] Created opam package (reason-react.opam)
- [0.11.0] Requires dune 3.8.0 (via dune-project)
- [0.11.0] Full migration to Melange

## Migration

First make sure your project builds with Melange, once this is done, you can follow the migration guide:

> - Install latest version of ReasonReact: `opam install reason-react`

Reviewer 2: ```suggestion
- Install the latest version of ReasonReact: `opam install reason-react`
```

> - Use this as a package in the dune file `(libraries reason-react)` (make sure to add `(preprocess (pps react-jsx-ppx))` as well).

Reviewer 2: ```suggestion
- In your dune file, add `reason-react` to `libraries` and preprocess with `reactjs-jsx-ppx`:

> (libraries reason-react)
> (preprocess (pps react-jsx-ppx)

```

> - Replace usages of modules with their newer names: `ReactDOMRe` -> `ReactDOM`, `ReactEventRe` -> `ReactEvent`, `ReasonReact` -> `React`.

Reviewer 2: ```suggestion
- Replace any previously deprecated modules with their new names: `ReactDOMRe` -> `ReactDOM`, `ReactEventRe` -> `ReactEvent`, `ReasonReact` -> `React`.
```

> - Remove usages of `ReasonReactCompat` and `ReasonReactOptimizedCreateClass`. They were deprecated a long time ago, and we removed them in this version.

Reviewer 2: ```suggestion
- Remove any usage of `ReasonReactCompat` and `ReasonReactOptimizedCreateClass`. These have long been deprecated, and have been removed in this version.
```

## Future

We started moving all ReasonReact related packages to Melange and 0.11 and want to make sure the ReasonML community can update the 3rd party libraries to use Melange as well.

> Some of them are published in [github.com/melange-community](https://github.com/melange-community/) or [github.com/ahrefs](https://github.com/ahrefs) such as [https://github.com/ahrefs/melange-react-dates](melange-react-dates) or [melange-recharts](https://github.com/ahrefs/melange-recharts).

Reviewer 2: ```suggestion
Some of them are published in [github.com/melange-community](https://github.com/melange-community/) or [github.com/ahrefs](https://github.com/ahrefs). Examples include [melange-react-dates](https://github.com/ahrefs/melange-react-dates) or [melange-recharts](https://github.com/ahrefs/melange-recharts).
```

> There's a lot of work to do for 0.12 work, but we are excited to continue to improve ReasonReact and make it a great library for the Melange and ReasonML community.

Reviewer 2: ```suggestion
There's lots to do for the next version. We are committed to continue improving ReasonReact and make it a great library for the Melange and ReasonML community. In the future, we expect to make the following changes:
```

> - Bring reacjs-jsx-ppx into this repository

Reviewer 2: ```suggestion
- Track `reacjs-jsx-ppx` in this repository, and rename it to `reason-react-ppx`, a more suitable name
```

> - Use the JSX transformation from React 17

Reviewer 2: ```suggestion
- Use the [new JSX transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) from React 17
```

- Fix code-generation locations for JSX and props

> - A bunch of documentation work

Reviewer 2: ```suggestion
- Continue improving the documentation
```

> Please contact us if you have any questions or want to help, joining [our Discord](https://discord.gg/reasonml) is the best way to reach us.

Reviewer 2: ```suggestion
Feel free to reach out if you have any feedback. You can [open an issue](https://github.com/reasonml/reason-react/issues) in this repository, or join the [ReasonML Discord](https://discord.gg/reasonml).
```

---
title: ReasonReact is back (and powered by Melange)
---

ReasonReact moved to `@rescript/react` after the ReScript announcement. The future of ReasonReact has been uncertain ever since. In the meantime, the Melange team announced the first major release of [Melange](https://anmonteiro.substack.com/p/melange-10-is-here) and took the first steps towards deeper integration with the OCaml ecosystem.

Today, we are happy to announce that ReasonReact is back and will be maintained by the Melange team going forward.

## What's changing?

> We released ReasonReact to the [opam repository](https://ocaml.org/p/reason-react/latest). In the Melange docs, we document [how to depend](https://melange.re/v1.0.0/package-management/) on packages in opam.

Reviewer: There's a newer section with more details in https://melange.re/unstable/package-management/#finding-and-using-melange-compatible-packages, maybe worth linking to it directly.

Author: I prefer to keep 1.0.0 stuff under this blog post
